ansible_collections/arista/avd/plugins/filter/password.py

```diff
@@ -1,9 +1,90 @@
 """
 Encrypt / Decrypt filters
 """
+from __future__ import annotations
 
-from ansible_collections.arista.avd.plugins.plugin_utils.bgp_utils import cbc_decrypt, cbc_encrypt
 from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError, AristaAvdMissingVariableError
+from ansible_collections.arista.avd.plugins.plugin_utils.password_utils import cbc_decrypt, cbc_encrypt
+
+
+##############
+# OSPF
+##############
+def ospf_encrypt(password: str, key: str, auth_algo: str | None = None, key_id: str | None = None) -> str:
+    """
+    Encrypt a password.
+
+    <key> should be the interface name e.g. "Ethernet1"
+
+    The key is transformed to either::
+        <key>_passwd for simple authentication
+    or
+        <key>_<auth_algo>Key_<key_id> for message digest keys
+
+    Returns the encrypted password as a string
+    """
+    if not key:
+        raise AristaAvdMissingVariableError("Key is required for OSPF encryption")
+
+    if not password:
+        raise AristaAvdMissingVariableError("Password is required for OSPF encryption")
+
+    if not isinstance(password, str):
+        raise AristaAvdError(f"Password MUST be of type 'str' but is of type {type(password)}")
+
+    data = bytes(password, encoding="UTF-8")
+
+    if auth_algo is not None and key_id is not None:
```

Comment:
Add validation of "auth_algo" and rename it to `hash_algorithm` to match the schema key